ot-wpantund: introduce OpenThread wpantund container

mike-scott opened the pull request:
This container starts the OpenThread wpantund daemon using
Zephyr-compatible defaults.

When paired with a nat64 service this container provides
border router access to an OpenThread network.

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott commented:
Ugh, wrong version of the start.sh script.  Fixing.

doanac commented:
Is there a simple command you can run that can serve as a simple test? eg something like this:

 https://github.com/foundriesio/gateway-containers/blob/master/bt-joiner/docker-build.conf

improves our CI a little

mike-scott commented:
@doanac ah, so when the container comes up it can run a test command to see if its working?

mike-scott commented:
@doanac added docker-build.conf

doanac commented:
thanks mike. LGTM from CI/release side

mike-scott commented on ot-wpantund/start.sh:
```diff
@@ -0,0 +1,127 @@
+sleep 2
+
+# Add border router
+wpanctl add-prefix -l 64 -r -s -f -a -o -P 1 "fd11:22::"
```
Ugh, forgot the parameter to set this

ricardosalveti commented on ot-wpantund/Dockerfile:
```diff
@@ -0,0 +1,65 @@
+LABEL maintainer="Michael Scott <mike@foundries.io>"
+
+# Add autoconf-archive
+RUN echo "@testing http://nl.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
```
Why is this required?

mike-scott replied:
autoconf-archive is only available in the testing repo of edge atm.

ricardosalveti commented on ot-wpantund/Dockerfile:
```diff
@@ -0,0 +1,65 @@
+
+RUN mkdir -p /dev/net && \
+    mknod /dev/net/tun c 10 200 && \
+    chmod 600 /dev/net/tun
```
mknod supports -m to set the permission as well, so chmod can be removed.

mike-scott replied:
fixing

ricardosalveti commented on ot-wpantund/start.sh:
```diff
@@ -0,0 +1,127 @@
+#!/bin/sh
+
```
Please add license and copyright.

mike-scott replied:
adding

ricardosalveti commented on ot-wpantund/start.sh:
```diff
@@ -0,0 +1,127 @@
+{
+    while [ $# -gt 0 ]
+    do
+        echo $1
```
Do we need this echo here or was it just for debugging?

ricardosalveti commented on ot-wpantund/start.sh:
```diff
@@ -0,0 +1,127 @@
+[ -n "$NCP_PATH" ] || NCP_PATH="/dev/ttyACM0"
+[ -n "$TUN_INTERFACE_NAME" ] || TUN_INTERFACE_NAME="wpan0"
+[ -n "$AUTO_PREFIX_ROUTE" ] || AUTO_PREFIX_ROUTE=true
+[ -n "$AUTO_PREFIX_SLAAC" ] || AUTO_PREFIX_SLAAC=true
```
Isn't it better to just set the defaults before parsing the argument instead of running a [ -n ] for every variable?

mike-scott replied:
fixing

ricardosalveti commented on ot-wpantund/start.sh:
```diff
@@ -0,0 +1,127 @@
+[ -n "$AUTO_PREFIX_SLAAC" ] || AUTO_PREFIX_SLAAC=true
+
+# Create tun node
+mkdir -p /dev/net && mknod /dev/net/tun c 10 200 && chmod 600 /dev/net/tun
```
This is also created in the dockerfile, do we need to duplicate it here as well?

mike-scott replied:
removing this one and testing again

doanac commented:
Note the README.md's are all following a similar convention these days. Eg: https://raw.githubusercontent.com/foundriesio/gateway-containers/master/bt-joiner/README.md

Roughly speaking its:
~~~
<Description possibly including hyperlinks to upstreams>

## How to use this image

```
docker run --privileged --net=host hub.foundries.io/bt-joiner
```
~~~

mike-scott commented:
- parameters for border router prefix settings
- license and copyrights
- README.md
- Addressed comments

I need to rebuild and run some tests before merging if everything looks ok here

mike-scott commented on ot-wpantund/Dockerfile:
```diff
@@ -0,0 +1,79 @@
+RUN apk add --no-cache libstdc++ libgcc dbus readline net-tools openrc
+
+RUN mkdir -p /dev/net && \
+    mknod --mode 600 /dev/net/tun c 10 200
```
Fixing this.  Busybox mknod doesn't support "--mode" only "-m"

mike-scott commented:
- changed permissions on start.sh and removed the chmod from Dockerfile
